[Spree Upgrade] Improve order state translations in v2

luisramos0 opened the pull request:
#### What? Why?

Closes #3542

Here we are improving the way we manage order state translations by using en.spree.order_state on haml code and js.admin.orders.order_state in angular (remember the objective with js translations set in #2721). See what translations were changed in "what to test" below.

#### What should we test?
We need to validate the places where we are changing translations.

In the orders list:
- the order states in the filters
- the name of the columns: Payment State, Shipment State, Completed At, Number, State, Customer E-Mail
- each value of the order state column
- each value of the shipment state column
- each value of the payments state column

In order edit page, the state of the order on the right side panel

In the subscriptions order panel, the orders state (this one can’t be tested yet in v2).

#### Dependencies
To be tested, this PR needs translations coming from master in #3535

sauloperez commented:
I added the dependency. Do you want to proceed with the testing as usual after #3535 gets merged?

luisramos0 commented:
yes, I think we can now switch to normal process for all v2 PRs with tests in staging

luisramos0 commented:
I rebased this to get all the translations from #3535

luisramos0 commented:
I manually put the latest **FR** translations from transifex on top of this PR and it's looking really good on my local but in AU staging there must be some cache of the JS translations and some are not showing up in French...
@kristinalim @sauloperez @mkllnk would you know what this is? JS translations not being picked up in the order states translations here: https://staging.openfoodnetwork.org.au/admin/orders

sigmundpetersen commented:
I've seen something like this before and I think there was some mention of asset precompile step on deploy. Do you know if this was performed? 

Maybe also a unicorn/nginx restart would be needed

mkllnk commented:
When we had issues with translations not appearing it was because not all assets got recompiled. It helped to delete `tmp/cache` but that won't help here. The server is now using ofn-install which creates a new directory during deploy, without `tmp`. The assets folder is shared though. Somehow it may think that the assets are already compiled. You can try to delete all compiled assets and then deploy again.

luisramos0 commented:
Master Maikel! Thank you. I just deleted everything from apps/openfoodnetwork/shared/assets and re-deployed, fixed :-)

This is ready for testing in staging AUS.

luisramos0 commented:
re canceled - I see there's a shipment_states.canceled entry missing, maybe that is causing the problem. I am not 100% sure. Anyway, it's better to create a separate issue as the fix will be a new translation key to master (needs a PR against master, this one is a v2 PR).

re other states - you can see the cart state by listing orders not completed. if you dont see any order in cart, just add an item to the cart on the frontoffice and you will see a new order in the backoffice.
I am not sure about the address and payment states, I think they will be seen when a shipping method validation or a payment fails respectively. Maybe this can be achieved in the backoffice by creating a payment that fails? I think this question deserves an issue, even if tech debt issue. we should clarify when would an order end up in one of these states...

RachL commented:
@luisramos0 I've just noticed that if you filter on "cart" the checkbox "show completed orders only" remains checked. So my guess is that I forgot to uncheck it during my tests. Let's move forward with this PR. I will create a new issue for the cancel button missing and if I see further translations missing in my tests, I will do the same.
